// @gear-js/vara-ui/dist/vara-ui.js
import * as Fe from "react";
import Ae, { forwardRef as z, useId as ie, useState as Er, useEffect as Rr } from "react";
import { createPortal as Cr } from "react-dom";
var le = { exports: {} }, M = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Le;
function jr() {
  if (Le)
    return M;
  Le = 1;
  var i = Ae, s = Symbol.for("react.element"), c = Symbol.for("react.fragment"), a = Object.prototype.hasOwnProperty, b = i.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, v = { key: !0, ref: !0, __self: !0, __source: !0 };
  function g(h, f, y) {
    var E, j = {}, P = null, q = null;
    y !== void 0 && (P = "" + y), f.key !== void 0 && (P = "" + f.key), f.ref !== void 0 && (q = f.ref);
    for (E in f)
      a.call(f, E) && !v.hasOwnProperty(E) && (j[E] = f[E]);
    if (h && h.defaultProps)
      for (E in f = h.defaultProps, f)
        j[E] === void 0 && (j[E] = f[E]);
    return { $$typeof: s, type: h, key: P, ref: q, props: j, _owner: b.current };
  }
  return M.Fragment = c, M.jsx = g, M.jsxs = g, M;
}
var V = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var De;
function Tr() {
  return De || (De = 1, process.env.NODE_ENV !== "production" && function() {
    var i = Ae, s = Symbol.for("react.element"), c = Symbol.for("react.portal"), a = Symbol.for("react.fragment"), b = Symbol.for("react.strict_mode"), v = Symbol.for("react.profiler"), g = Symbol.for("react.provider"), h = Symbol.for("react.context"), f = Symbol.for("react.forward_ref"), y = Symbol.for("react.suspense"), E = Symbol.for("react.suspense_list"), j = Symbol.for("react.memo"), P = Symbol.for("react.lazy"), q = Symbol.for("react.offscreen"), ce = Symbol.iterator, Ye = "@@iterator";
    function Me(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = ce && e[ce] || e[Ye];
      return typeof r == "function" ? r : null;
    }
    var F = i.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function R(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), n = 1; n < r; n++)
          t[n - 1] = arguments[n];
        Ve("error", e, t);
      }
    }
    function Ve(e, r, t) {
      {
        var n = F.ReactDebugCurrentFrame, d = n.getStackAddendum();
        d !== "" && (r += "%s", t = t.concat([d]));
        var _ = t.map(function(u) {
          return String(u);
        });
        _.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, _);
      }
    }
    var Ue = !1, Be = !1, qe = !1, Je = !1, Ge = !1, ue;
    ue = Symbol.for("react.module.reference");
    function Ke(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === a || e === v || Ge || e === b || e === y || e === E || Je || e === q || Ue || Be || qe || typeof e == "object" && e !== null && (e.$$typeof === P || e.$$typeof === j || e.$$typeof === g || e.$$typeof === h || e.$$typeof === f || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === ue || e.getModuleId !== void 0));
    }
    function He(e, r, t) {
      var n = e.displayName;
      if (n)
        return n;
      var d = r.displayName || r.name || "";
      return d !== "" ? t + "(" + d + ")" : t;
    }
    function fe(e) {
      return e.displayName || "Context";
    }
    function O(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && R("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case a:
          return "Fragment";
        case c:
          return "Portal";
        case v:
          return "Profiler";
        case b:
          return "StrictMode";
        case y:
          return "Suspense";
        case E:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case h:
            var r = e;
            return fe(r) + ".Consumer";
          case g:
            var t = e;
            return fe(t._context) + ".Provider";
          case f:
            return He(e, e.render, "ForwardRef");
          case j:
            var n = e.displayName || null;
            return n !== null ? n : O(e.type) || "Memo";
          case P: {
            var d = e, _ = d._payload, u = d._init;
            try {
              return O(u(_));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var N = Object.assign, W = 0, de, _e, be, ve, me, ge, pe;
    function he() {
    }
    he.__reactDisabledLog = !0;
    function Xe() {
      {
        if (W === 0) {
          de = console.log, _e = console.info, be = console.warn, ve = console.error, me = console.group, ge = console.groupCollapsed, pe = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: he,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        W++;
      }
    }
    function Ze() {
      {
        if (W--, W === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: N({}, e, {
              value: de
            }),
            info: N({}, e, {
              value: _e
            }),
            warn: N({}, e, {
              value: be
            }),
            error: N({}, e, {
              value: ve
            }),
            group: N({}, e, {
              value: me
            }),
            groupCollapsed: N({}, e, {
              value: ge
            }),
            groupEnd: N({}, e, {
              value: pe
            })
          });
        }
        W < 0 && R("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var Z = F.ReactCurrentDispatcher, Q;
    function J(e, r, t) {
      {
        if (Q === void 0)
          try {
            throw Error();
          } catch (d) {
            var n = d.stack.trim().match(/\n( *(at )?)/);
            Q = n && n[1] || "";
          }
        return `
` + Q + e;
      }
    }
    var ee = !1, G;
    {
      var Qe = typeof WeakMap == "function" ? WeakMap : Map;
      G = new Qe();
    }
    function ye(e, r) {
      if (!e || ee)
        return "";
      {
        var t = G.get(e);
        if (t !== void 0)
          return t;
      }
      var n;
      ee = !0;
      var d = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var _;
      _ = Z.current, Z.current = null, Xe();
      try {
        if (r) {
          var u = function() {
            throw Error();
          };
          if (Object.defineProperty(u.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(u, []);
            } catch (k) {
              n = k;
            }
            Reflect.construct(e, [], u);
          } else {
            try {
              u.call();
            } catch (k) {
              n = k;
            }
            e.call(u.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (k) {
            n = k;
          }
          e();
        }
      } catch (k) {
        if (k && n && typeof k.stack == "string") {
          for (var l = k.stack.split(`
`), C = n.stack.split(`
`), m = l.length - 1, p = C.length - 1; m >= 1 && p >= 0 && l[m] !== C[p]; )
            p--;
          for (; m >= 1 && p >= 0; m--, p--)
            if (l[m] !== C[p]) {
              if (m !== 1 || p !== 1)
                do
                  if (m--, p--, p < 0 || l[m] !== C[p]) {
                    var T = `
` + l[m].replace(" at new ", " at ");
                    return e.displayName && T.includes("<anonymous>") && (T = T.replace("<anonymous>", e.displayName)), typeof e == "function" && G.set(e, T), T;
                  }
                while (m >= 1 && p >= 0);
              break;
            }
        }
      } finally {
        ee = !1, Z.current = _, Ze(), Error.prepareStackTrace = d;
      }
      var D = e ? e.displayName || e.name : "", Ne = D ? J(D) : "";
      return typeof e == "function" && G.set(e, Ne), Ne;
    }
    function er(e, r, t) {
      return ye(e, !1);
    }
    function rr(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function K(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return ye(e, rr(e));
      if (typeof e == "string")
        return J(e);
      switch (e) {
        case y:
          return J("Suspense");
        case E:
          return J("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case f:
            return er(e.render);
          case j:
            return K(e.type, r, t);
          case P: {
            var n = e, d = n._payload, _ = n._init;
            try {
              return K(_(d), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var H = Object.prototype.hasOwnProperty, xe = {}, Ee = F.ReactDebugCurrentFrame;
    function X(e) {
      if (e) {
        var r = e._owner, t = K(e.type, e._source, r ? r.type : null);
        Ee.setExtraStackFrame(t);
      } else
        Ee.setExtraStackFrame(null);
    }
    function tr(e, r, t, n, d) {
      {
        var _ = Function.call.bind(H);
        for (var u in e)
          if (_(e, u)) {
            var l = void 0;
            try {
              if (typeof e[u] != "function") {
                var C = Error((n || "React class") + ": " + t + " type `" + u + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[u] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw C.name = "Invariant Violation", C;
              }
              l = e[u](r, u, n, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (m) {
              l = m;
            }
            l && !(l instanceof Error) && (X(d), R("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", n || "React class", t, u, typeof l), X(null)), l instanceof Error && !(l.message in xe) && (xe[l.message] = !0, X(d), R("Failed %s type: %s", t, l.message), X(null));
          }
      }
    }
    var nr = Array.isArray;
    function re(e) {
      return nr(e);
    }
    function ar(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function sr(e) {
      try {
        return Re(e), !1;
      } catch {
        return !0;
      }
    }
    function Re(e) {
      return "" + e;
    }
    function Ce(e) {
      if (sr(e))
        return R("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", ar(e)), Re(e);
    }
    var Y = F.ReactCurrentOwner, or = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, je, Te, te;
    te = {};
    function lr(e) {
      if (H.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function ir(e) {
      if (H.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function cr(e, r) {
      if (typeof e.ref == "string" && Y.current && r && Y.current.stateNode !== r) {
        var t = O(Y.current.type);
        te[t] || (R('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', O(Y.current.type), e.ref), te[t] = !0);
      }
    }
    function ur(e, r) {
      {
        var t = function() {
          je || (je = !0, R("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function fr(e, r) {
      {
        var t = function() {
          Te || (Te = !0, R("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var dr = function(e, r, t, n, d, _, u) {
      var l = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: s,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: t,
        props: u,
        // Record the component responsible for creating this element.
        _owner: _
      };
      return l._store = {}, Object.defineProperty(l._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(l, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: n
      }), Object.defineProperty(l, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: d
      }), Object.freeze && (Object.freeze(l.props), Object.freeze(l)), l;
    };
    function _r(e, r, t, n, d) {
      {
        var _, u = {}, l = null, C = null;
        t !== void 0 && (Ce(t), l = "" + t), ir(r) && (Ce(r.key), l = "" + r.key), lr(r) && (C = r.ref, cr(r, d));
        for (_ in r)
          H.call(r, _) && !or.hasOwnProperty(_) && (u[_] = r[_]);
        if (e && e.defaultProps) {
          var m = e.defaultProps;
          for (_ in m)
            u[_] === void 0 && (u[_] = m[_]);
        }
        if (l || C) {
          var p = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          l && ur(u, p), C && fr(u, p);
        }
        return dr(e, l, C, d, n, Y.current, u);
      }
    }
    var ne = F.ReactCurrentOwner, Se = F.ReactDebugCurrentFrame;
    function L(e) {
      if (e) {
        var r = e._owner, t = K(e.type, e._source, r ? r.type : null);
        Se.setExtraStackFrame(t);
      } else
        Se.setExtraStackFrame(null);
    }
    var ae;
    ae = !1;
    function se(e) {
      return typeof e == "object" && e !== null && e.$$typeof === s;
    }
    function we() {
      {
        if (ne.current) {
          var e = O(ne.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function br(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var $e = {};
    function vr(e) {
      {
        var r = we();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function Oe(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = vr(r);
        if ($e[t])
          return;
        $e[t] = !0;
        var n = "";
        e && e._owner && e._owner !== ne.current && (n = " It was passed a child from " + O(e._owner.type) + "."), L(e), R('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, n), L(null);
      }
    }
    function ke(e, r) {
      {
        if (typeof e != "object")
          return;
        if (re(e))
          for (var t = 0; t < e.length; t++) {
            var n = e[t];
            se(n) && Oe(n, r);
          }
        else if (se(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var d = Me(e);
          if (typeof d == "function" && d !== e.entries)
            for (var _ = d.call(e), u; !(u = _.next()).done; )
              se(u.value) && Oe(u.value, r);
        }
      }
    }
    function mr(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === f || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === j))
          t = r.propTypes;
        else
          return;
        if (t) {
          var n = O(r);
          tr(t, e.props, "prop", n, e);
        } else if (r.PropTypes !== void 0 && !ae) {
          ae = !0;
          var d = O(r);
          R("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", d || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && R("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function gr(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var n = r[t];
          if (n !== "children" && n !== "key") {
            L(e), R("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", n), L(null);
            break;
          }
        }
        e.ref !== null && (L(e), R("Invalid attribute `ref` supplied to `React.Fragment`."), L(null));
      }
    }
    function Pe(e, r, t, n, d, _) {
      {
        var u = Ke(e);
        if (!u) {
          var l = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (l += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var C = br(d);
          C ? l += C : l += we();
          var m;
          e === null ? m = "null" : re(e) ? m = "array" : e !== void 0 && e.$$typeof === s ? (m = "<" + (O(e.type) || "Unknown") + " />", l = " Did you accidentally export a JSX literal instead of a component?") : m = typeof e, R("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", m, l);
        }
        var p = _r(e, r, t, d, _);
        if (p == null)
          return p;
        if (u) {
          var T = r.children;
          if (T !== void 0)
            if (n)
              if (re(T)) {
                for (var D = 0; D < T.length; D++)
                  ke(T[D], e);
                Object.freeze && Object.freeze(T);
              } else
                R("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              ke(T, e);
        }
        return e === a ? gr(p) : mr(p), p;
      }
    }
    function pr(e, r, t) {
      return Pe(e, r, t, !0);
    }
    function hr(e, r, t) {
      return Pe(e, r, t, !1);
    }
    var yr = hr, xr = pr;
    V.Fragment = a, V.jsx = yr, V.jsxs = xr;
  }()), V;
}
process.env.NODE_ENV === "production" ? le.exports = jr() : le.exports = Tr();
var o = le.exports;
function Ie(i) {
  var s, c, a = "";
  if (typeof i == "string" || typeof i == "number")
    a += i;
  else if (typeof i == "object")
    if (Array.isArray(i))
      for (s = 0; s < i.length; s++)
        i[s] && (c = Ie(i[s])) && (a && (a += " "), a += c);
    else
      for (s in i)
        i[s] && (a && (a += " "), a += s);
  return a;
}
function x() {
  for (var i, s, c = 0, a = ""; c < arguments.length; )
    (i = arguments[c++]) && (s = Ie(i)) && (a && (a += " "), a += s);
  return a;
}
const Sr = "_button_xgtmc_11", wr = "_blink_xgtmc_1", $r = "_disabled_xgtmc_31", Or = "_loading_xgtmc_35", kr = "_primary_xgtmc_46", Pr = "_dark_xgtmc_65", Nr = "_light_xgtmc_85", Fr = "_border_xgtmc_103", Lr = "_transparent_xgtmc_118", Dr = "_noText_xgtmc_130", Ar = "_small_xgtmc_136", A = {
  button: Sr,
  blink: wr,
  disabled: $r,
  loading: Or,
  primary: kr,
  dark: Pr,
  light: Nr,
  border: Fr,
  transparent: Lr,
  default: "_default_xgtmc_126",
  noText: Dr,
  small: Ar
}, ze = z((i, s) => {
  const {
    className: c,
    text: a,
    icon: b,
    disabled: v,
    isLoading: g,
    type: h = "button",
    color: f = "primary",
    size: y = "default",
    children: E,
    ...j
  } = i;
  return /* @__PURE__ */ o.jsxs(
    "button",
    {
      type: h,
      className: x(
        A.button,
        A[f],
        f !== "transparent" && A[y],
        v && A.disabled,
        g && A.loading,
        !a && A.noText,
        c
      ),
      disabled: v || g,
      ref: s,
      ...j,
      children: [
        b && /* @__PURE__ */ o.jsx(b, {}),
        a && /* @__PURE__ */ o.jsx("span", { children: a }),
        E
      ]
    }
  );
}), Ir = "_root_zlz4l_1", zr = "_disabled_zlz4l_2", Wr = "_base_zlz4l_8", Yr = "_input_zlz4l_12", Mr = "_label_zlz4l_13", Vr = "_small_zlz4l_23", Ur = "_error_zlz4l_45", Br = "_fieldset_zlz4l_50", qr = "_legendLabel_zlz4l_51", Jr = "_legend_zlz4l_51", Gr = "_message_zlz4l_118", S = {
  root: Ir,
  disabled: zr,
  base: Wr,
  input: Yr,
  label: Mr,
  default: "_default_zlz4l_19",
  small: Vr,
  error: Ur,
  fieldset: Br,
  legendLabel: qr,
  legend: Jr,
  message: Gr
}, Mt = z(
  ({ className: i, label: s, error: c, type: a = "text", placeholder: b = " ", size: v = "default", ...g }, h) => {
    const { disabled: f } = g, y = ie();
    return /* @__PURE__ */ o.jsxs("div", { className: x(S.root, i, f && S.disabled), children: [
      /* @__PURE__ */ o.jsxs("div", { className: S.base, children: [
        /* @__PURE__ */ o.jsx(
          "input",
          {
            type: a,
            id: y,
            className: x(S.input, S[v], c && S.error),
            placeholder: b,
            ref: h,
            ...g
          }
        ),
        s && /* @__PURE__ */ o.jsx("label", { htmlFor: y, className: x(S.label, S[v]), children: s }),
        /* @__PURE__ */ o.jsx("fieldset", { className: S.fieldset, children: /* @__PURE__ */ o.jsxs("legend", { className: x(S.legend, s && S.legendLabel), children: [
          s,
          "​"
        ] }) })
      ] }),
      c && /* @__PURE__ */ o.jsx("p", { className: S.message, children: c })
    ] });
  }
), Kr = "_root_f7rox_1", Hr = "_disabled_f7rox_2", Xr = "_base_f7rox_8", Zr = "_textarea_f7rox_12", Qr = "_label_f7rox_13", et = "_small_f7rox_23", rt = "_error_f7rox_46", tt = "_fieldset_f7rox_51", nt = "_legendLabel_f7rox_52", at = "_legend_f7rox_52", st = "_message_f7rox_125", w = {
  root: Kr,
  disabled: Hr,
  base: Xr,
  textarea: Zr,
  label: Qr,
  default: "_default_f7rox_19",
  small: et,
  error: rt,
  fieldset: tt,
  legendLabel: nt,
  legend: at,
  message: st
}, Vt = z(
  ({ className: i, label: s, error: c, size: a = "default", rows: b = 5, placeholder: v = " ", ...g }, h) => {
    const { disabled: f } = g, y = ie();
    return /* @__PURE__ */ o.jsxs("div", { className: x(w.root, i, f && w.disabled), children: [
      /* @__PURE__ */ o.jsxs("div", { className: w.base, children: [
        /* @__PURE__ */ o.jsx(
          "textarea",
          {
            rows: b,
            id: y,
            className: x(w.textarea, w[a], c && w.error),
            placeholder: v,
            ref: h,
            ...g
          }
        ),
        s && /* @__PURE__ */ o.jsx("label", { htmlFor: y, className: x(w.label, w[a]), children: s }),
        /* @__PURE__ */ o.jsx("fieldset", { className: w.fieldset, children: /* @__PURE__ */ o.jsxs("legend", { className: x(w.legend, s && w.legendLabel), children: [
          s,
          "​"
        ] }) })
      ] }),
      c && /* @__PURE__ */ o.jsx("p", { className: w.message, children: c })
    ] });
  }
), ot = "_root_93mfb_1", lt = "_disabled_93mfb_2", it = "_base_93mfb_8", ct = "_select_93mfb_12", ut = "_label_93mfb_13", ft = "_small_93mfb_24", dt = "_error_93mfb_44", _t = "_fieldset_93mfb_49", bt = "_legendLabel_93mfb_50", vt = "_legend_93mfb_50", mt = "_message_93mfb_116", $ = {
  root: ot,
  disabled: lt,
  base: it,
  select: ct,
  label: ut,
  default: "_default_93mfb_20",
  small: ft,
  error: dt,
  fieldset: _t,
  legendLabel: bt,
  legend: vt,
  message: mt
}, Ut = z(
  ({ options: i, className: s, label: c, error: a, size: b = "default", ...v }, g) => {
    const { disabled: h } = v, f = ie(), y = () => i.map((E, j) => /* @__PURE__ */ o.jsx("option", { ...E }, j));
    return /* @__PURE__ */ o.jsxs("div", { className: x($.root, s, h && $.disabled), children: [
      /* @__PURE__ */ o.jsxs("div", { className: $.base, children: [
        /* @__PURE__ */ o.jsx("select", { id: f, className: x($.select, $[b], a && $.error), ref: g, ...v, children: y() }),
        c && /* @__PURE__ */ o.jsx("label", { htmlFor: f, className: x($.label, $[b]), children: c }),
        /* @__PURE__ */ o.jsx("fieldset", { className: $.fieldset, children: /* @__PURE__ */ o.jsxs("legend", { className: x($.legend, c && $.legendLabel), children: [
          c,
          "​"
        ] }) })
      ] }),
      a && /* @__PURE__ */ o.jsx("p", { className: $.message, children: a })
    ] });
  }
), gt = "_label_17s34_1", pt = "_disabled_17s34_6", ht = "_input_17s34_12", yt = "_checkbox_17s34_18", U = {
  label: gt,
  disabled: pt,
  input: ht,
  checkbox: yt,
  switch: "_switch_17s34_28"
}, Bt = z(({ label: i, className: s, type: c, ...a }, b) => {
  const { disabled: v } = a;
  return /* @__PURE__ */ o.jsxs("label", { className: x(U.label, s, v && U.disabled), children: [
    /* @__PURE__ */ o.jsx(
      "input",
      {
        type: "checkbox",
        className: x(U.input, c === "switch" ? U.switch : U.checkbox),
        ref: b,
        ...a
      }
    ),
    i
  ] });
}), xt = "_label_12j8l_1", Et = "_disabled_12j8l_6", Rt = "_input_12j8l_12", oe = {
  label: xt,
  disabled: Et,
  input: Rt
}, qt = z(({ label: i, className: s, ...c }, a) => {
  const { disabled: b } = c;
  return /* @__PURE__ */ o.jsxs("label", { className: x(oe.label, s, b && oe.disabled), children: [
    /* @__PURE__ */ o.jsx("input", { type: "radio", className: oe.input, ref: a, ...c }),
    i
  ] });
}), We = (i) => /* @__PURE__ */ Fe.createElement("svg", { width: 16, height: 16, viewBox: "0 0 16 16", fill: "none", xmlns: "http://www.w3.org/2000/svg", ...i }, /* @__PURE__ */ Fe.createElement("path", { d: "M1.33454 0.54405C1.16856 0.544071 1.00638 0.593719 0.868832 0.686604C0.731282 0.779488 0.624641 0.911362 0.562608 1.06531C0.500576 1.21925 0.485983 1.38827 0.520706 1.55057C0.555429 1.71287 0.637881 1.86103 0.757471 1.97612L6.82396 8.04259L0.757471 14.1092C0.6774 14.1856 0.613391 14.2773 0.569192 14.3788C0.524994 14.4803 0.501494 14.5896 0.500069 14.7003C0.498644 14.811 0.519322 14.9209 0.560893 15.0235C0.602464 15.1261 0.664092 15.2195 0.742169 15.298C0.820245 15.3765 0.9132 15.4386 1.01559 15.4807C1.11798 15.5229 1.22775 15.5442 1.33847 15.5434C1.44919 15.5426 1.55864 15.5197 1.6604 15.476C1.76216 15.4324 1.85419 15.3689 1.93111 15.2892L8.00247 9.22113L14.0706 15.2892C14.148 15.3669 14.24 15.4286 14.3413 15.4707C14.4426 15.5128 14.5512 15.5345 14.6609 15.5346C14.7706 15.5347 14.8793 15.5132 14.9806 15.4712C15.082 15.4293 15.1741 15.3678 15.2517 15.2902C15.3292 15.2126 15.3908 15.1206 15.4327 15.0192C15.4746 14.9178 15.4962 14.8091 15.4961 14.6994C15.496 14.5897 15.4743 14.4812 15.4322 14.3799C15.3901 14.2786 15.3284 14.1866 15.2507 14.1092L9.18099 8.04259L15.2507 1.97612C15.3724 1.85924 15.4557 1.70819 15.4896 1.54289C15.5235 1.37759 15.5063 1.20584 15.4403 1.05052C15.3744 0.895208 15.2628 0.763621 15.1204 0.673164C14.978 0.582707 14.8114 0.537686 14.6428 0.54405C14.4266 0.550493 14.2213 0.64089 14.0706 0.796039L8.00247 6.86414L1.93111 0.796039C1.85355 0.716311 1.76079 0.652891 1.65833 0.609615C1.55586 0.566339 1.44576 0.544057 1.33454 0.54405Z", fill: "black" })), Ct = "_root_8054h_1", jt = "_alert_8054h_9", Tt = "_header_8054h_16", St = "_success_8054h_37", wt = "_error_8054h_45", $t = "_info_8054h_53", Ot = "_loading_8054h_61", kt = "_rotating_8054h_1", Pt = "_button_8054h_70", Nt = "_body_8054h_77", B = {
  root: Ct,
  alert: jt,
  header: Tt,
  success: St,
  error: wt,
  info: $t,
  loading: Ot,
  rotating: kt,
  button: Pt,
  body: Nt
};
function Jt({ alert: i, close: s }) {
  const { content: c, options: a } = i, { type: b, title: v, style: g, isClosed: h } = a;
  return /* @__PURE__ */ o.jsxs("div", { className: B.alert, style: g, children: [
    /* @__PURE__ */ o.jsxs("header", { className: x(B.header, B[b]), children: [
      v || b,
      h && /* @__PURE__ */ o.jsx(ze, { icon: We, color: "transparent", className: B.button, onClick: s })
    ] }),
    /* @__PURE__ */ o.jsx("div", { className: B.body, children: c })
  ] });
}
const Ft = "_overlay_smg58_1", Lt = "_modal_smg58_16", Dt = "_header_smg58_28", At = "_heading_smg58_34", It = "_button_smg58_45", zt = "_body_smg58_49", I = {
  overlay: Ft,
  modal: Lt,
  header: Dt,
  heading: At,
  button: It,
  body: zt
}, Gt = ({ heading: i, close: s, children: c, className: a }) => {
  const [b, v] = Er(), g = ({ target: f, currentTarget: y }) => {
    f === y && s();
  };
  Rr(() => {
    const f = document.createElement("div");
    return f.id = "modal-root", document.body.appendChild(f), v(f), () => {
      document.body.removeChild(f);
    };
  }, []);
  const h = /* @__PURE__ */ o.jsx("div", { className: I.overlay, onClick: g, children: /* @__PURE__ */ o.jsxs("div", { className: I.modal, children: [
    /* @__PURE__ */ o.jsxs("header", { className: I.header, children: [
      /* @__PURE__ */ o.jsx("h3", { className: I.heading, children: i }),
      /* @__PURE__ */ o.jsx(ze, { icon: We, color: "transparent", onClick: s, className: I.button })
    ] }),
    c && /* @__PURE__ */ o.jsx("div", { className: x(I.body, a), children: c })
  ] }) });
  return b ? Cr(h, b) : null;
};
export {
  Jt as Alert,
  ze as Button,
  Bt as Checkbox,
  Mt as Input,
  Gt as Modal,
  qt as Radio,
  Ut as Select,
  Vt as Textarea,
  B as alertStyles,
  A as buttonStyles,
  U as checkboxStyles,
  S as inputStyles,
  I as modalStyles,
  oe as radioStyles,
  $ as selectStyles,
  w as textareaStyles
};
